carp.data.core/src/commonTest/kotlin/dk/cachet/carp/data/infrastructure/StudyDataServiceDecoratorTest.kt

```diff
@@ -0,0 +1,52 @@
+package dk.cachet.carp.data.infrastructure
+
+import dk.cachet.carp.common.application.UUID
+import dk.cachet.carp.data.application.CollectedDataQuery
+import dk.cachet.carp.data.application.CollectedDataSet
+import dk.cachet.carp.data.application.StudyDataService
+import kotlinx.coroutines.test.runTest
+import kotlinx.datetime.Instant
+import kotlin.test.*
+
+class StudyDataServiceDecoratorTest
+{
+
+    private val studyId = UUID.randomUUID()
+    private val now = Instant.parse("2025-04-27T00:00:00Z")
+
+    private class TestStudyDataService : StudyDataService
+    {
+        var calledWith: StudyDataServiceRequest.GetCollectedData? = null
+
+        override suspend fun getCollectedData(
+            studyId: UUID,
+            query: CollectedDataQuery
+        ): CollectedDataSet
+        {
+            calledWith = StudyDataServiceRequest.GetCollectedData(
+                studyId, query
+            )
+            return CollectedDataSet() // Return empty for now
+        }
+    }
+
+    @Test
+    fun testGetCollectedDataIsDelegated() = runTest {
```

Comment:
Use underscores in test names (see all other tests).

And, no point starting every test with "test". Follow the structure "method_....." so that at a glance you can identify the SUT